src/KeyRegistry.sol

```diff
@@ -404,6 +422,18 @@ contract KeyRegistry is TrustedCaller, Signatures, EIP712, Nonces {
                                  ADMIN
     //////////////////////////////////////////////////////////////*/
 
+    /**
+     * @notice Set the IdRegistry contract address. Only callable by owner.
+     *
+     * @param scheme    The key scheme associated with this validator.
+     * @param typeId    The metadata typeId associated with this validator.
+     * @param validator Contract implementing IMetadataValidator.
+     */
+    function setValidator(uint32 scheme, uint8 typeId, IMetadataValidator validator) external onlyOwner {
```

Comment:
nit: should we allow scheme == 0 and type == 0?

Reply:
Yeah, we can catch these and revert if they're zero.